roles/opcap_tool/tasks/build.yml

```diff
@@ -15,12 +15,15 @@
     name: go-builder
     image: "golang:1.19"
     debug: true
-    pids_limit: -1
+    pids_limit: "-1"
     log_level: "info"
     command: make -C /opcap build OPCAP_VERSION={{ opcap_version }}
     detach: false
     tty: true
     rm: true
+    # Increasing the memory limit from 512m to 1g
+    # to fix recurrent 137 OOM SIGKILL during the opcap build
+    memory: "1g"
```

Comment:
Nice finding! I did not found where this OOM error was coming from

Reply:
Just for curiosity, where did you find that the limit was 512m? I was searching online again but I couldn't find it 🤕,  in some podman-run references I saw that if no memory is specified then the container’s memory is not limited

Reply:
I'm sorry but I'm not sure where the 512m limit was set, I looked at the docs but I couldn't find a default value, so I'm not sure how the 512 limit was found.

Correct me if I'm wrong, but the change suggest setting a limit rather than increasing it.

[1] https://docs.ansible.com/ansible/latest/collections/containers/podman/podman_container_module.html#parameter-memory
[2] https://docs.podman.io/en/latest/markdown/podman-run.1.html#memory-m-number-unit

Reply:
Sorry, I put all the details in the [JIRA ticket](https://issues.redhat.com/browse/CILAB-1363). 

> where did you find that the limit was 512m

I did not find about the podman, only about the [docker](https://medium.com/codex/how-to-limit-cpu-and-memory-usage-in-docker-containers-a024b429f55e), where it's indeed 512m.

Reply:
@manurodriguez I have measured the memory usage locally with `top` while running opcap build task:
```
# Before the tests
MiB Mem :  31838.3 total,   1063.8 free,  23522.4 used,   7252.2 buff/cache

# During the opcap build
MiB Mem :  31838.3 total,    721.3 free,  23864.4 used,   7252.6 buff/cache
```

23864.4 used - 23522.4 used = about 300MiB.

For sure, the estimation is really rough, `top` usage + locally I ran opcap build with `podman run`  instead of `podman_container` module. But overall, looks reasonable, and justifies the idea to move up with 1g.

Reply:
I am also not sure if there even was a limit before, by setting a limit now to 1Gb (which is reasonable) you're effectively containing the opcap run and that protecting the host system from consuming all its memory and being OOM-killed. Maybe that's what was happening before?

Reply:
@thekad Maybe, that's totally possible. 

There are also [more layers](https://issues.redhat.com/browse/CILAB-1363) to the story: cgroups v2 vs cgroups v1. Currently, we have cgroup v1 on our Dallas jumphost:

- cgroup v1 does not support resource claims from rootless containers
- cgroup v1 [is not supported by podman guys](https://github.com/containers/podman/issues/17582) anymore

So if speak about a good solid fix, we should probably first migrate to cgroup v2. But I'm scared even to estimate possible consequences :)